tools/infrastructure/api_compare.py

```diff
@@ -0,0 +1,487 @@
+    for rpc_name in sorted(summary_result.keys()):
+        print "\n" + "---" * 60
+        print colors.HEADER + rpc_name + colors.ENDC
+        for problematic_item in summary_result[rpc_name]:
+            item = summary_result[rpc_name][problematic_item]
+            if len(item) > 0:
+                print colors.UNDERLINE + problematic_item + colors.ENDC
+            if type(item) is not dict:
+                print("{}{}{}".format(colors.WARN, item, colors.ENDC))
+            elif type(item) is dict:
+                for param in item.keys():
+                    item_print = colors.UNDERLINE + param + colors.ENDC
+                    print "{} {}".format("Parameter name: ", item_print)
+                    res_val = item[param]
+                    for key in res_val:
+                        print key, ":", colors.FAIL, res_val[key], colors.ENDC
+
+
+def print_summary_info(summary_result, args):
+    """Function which prints summary comparison info(if any).
+    Output type depends on command line args
+    """
+    summary_color = colors.UNDERLINE + colors.BOLD + colors.BLUE
+    print "\n"
+    print summary_color, "SUMMARY COMPARISON RESULT:\n", colors.ENDC
+
+    if len(summary_result) == 0:
+        print colors.BOLD + " === NO PROBLEMS FOUND ===" + colors.ENDC
+        return
+
+    if args.output == "console":
+        console_print(summary_result)
+    if args.output == "json":
+        json_summary_result = dict_to_json(summary_result)
+        print json_summary_result
+    if args.output == "xml":
+        json_summary_result = dict_to_json(summary_result)
+        temp = json.loads(json_summary_result)
+        xml_summary_result = json_to_xml(temp)
+        print xml_summary_result
+
+
+def handle_hmi_absent_params(hmi_absent_params, rpc_name, summary_result):
+    """Function which writes hmi absent parameters(if any)
+    to summary result
+    """
+    if len(hmi_absent_params) > 0:
+        fill_summary_result(
+            rpc_name, "Missed in HMI APIs:",
+            hmi_absent_params,
+            summary_result
+        )
+
+
+def handle_mob_absent_params(mob_absent_params, rpc_name, summary_result):
+    """Function which writes mobile absent parameters(if any)
+    to summary result
+    """
+    if len(mob_absent_params) > 0:
+        fill_summary_result(
+            rpc_name, "Missed in Mobile APIs:",
+            mob_absent_params,
+            summary_result
+        )
+
+
+def get_absent_hmi_params(hmi_params, mob_params, rpc):
+    """Function caculates absent parameters in hmi api"""
+    hmi_absent_params = set(
+        mob_params.keys()) - set(hmi_params.keys())
+    hmi_absent_params = filter_params(
+        rpc, hmi_absent_params,
+        exclude_redundant_from_hmi_rpc_filter
+    )
+    return hmi_absent_params
+
+
+def get_absent_mob_params(hmi_params, mob_params, rpc):
+    """Function caculates absent parameters in mobile api"""
+    mob_absent_params = list(
```

Comment:
@AByzhynar can you combine this functions too?
Also, why mob_absent_params is list but hmi_absent_params  is set?